Userland/Libraries/LibAccelGfx/Painter.cpp

```diff
@@ -690,4 +705,27 @@ void Painter::blit_scaled_texture(Gfx::FloatRect const& dst_rect, GL::Texture co
     GL::delete_vertex_array(vao);
 }
 
+void Painter::update_immutable_bitmap_texture_cache(Vector<Gfx::ImmutableBitmap const&>& immutable_bitmaps)
+{
+    for (auto immutable_bitmap_id : s_immutable_bitmap_texture_cache.keys()) {
+        auto it = immutable_bitmaps.find_if([&](auto const& immutable_bitmap) {
```

Comment:
Can we make `immutable_bitmaps` also be a `HashMap` or `HashTable` here?
Going on an `O(n)` side quest through a `Vector` for every bitmap seems suboptimal.

Reply:
makes sense. I changed it to use `HashMap`.